ql/src/java/org/apache/hadoop/hive/ql/exec/tez/monitoring/TezJobMonitor.java

```diff
@@ -124,17 +124,17 @@ public TezJobMonitor(List<BaseWork> topSortedWorks, final DAGClient dagClient, H
     this.dag = dag;
     this.context = ctx;
     console = SessionState.getConsole();
+    this.perfLogger = perfLogger;
```

Comment:
Why move this line of code?

Reply:
Since it is using here, it needs to be set before the `updateFunction` is called.
https://github.com/apache/hive/blob/82099b8fbf7e7ec3f3aaa778ddd70b8f959c1ad2/ql/src/java/org/apache/hadoop/hive/ql/exec/tez/monitoring/TezJobMonitor.java#L136-L137

https://github.com/apache/hive/blob/82099b8fbf7e7ec3f3aaa778ddd70b8f959c1ad2/ql/src/java/org/apache/hadoop/hive/ql/exec/tez/monitoring/TezJobMonitor.java#L128